diff --git a/tpls/mdspan/include/experimental/__p0009_bits/macros.hpp b/tpls/mdspan/include/experimental/__p0009_bits/macros.hpp
index b60c42617..dd69bd61e 100644
--- a/tpls/mdspan/include/experimental/__p0009_bits/macros.hpp
+++ b/tpls/mdspan/include/experimental/__p0009_bits/macros.hpp
@@ -116,6 +116,7 @@ MDSPAN_FUNCTION inline void default_precondition_violation_handler(const char* c
   assert(0);
 }
 #elif defined(_MDSPAN_HAS_SYCL)
+#include "sycl/sycl.hpp"
 MDSPAN_FUNCTION inline void default_precondition_violation_handler(const char* cond, const char* file, unsigned line)
 {
   sycl::ext::oneapi::experimental::printf("%s:%u: precondition failure: `%s`\n", file, line, cond);
