diff --git a/CMakeLists.txt b/CMakeLists.txt
index cfa1c6f..3f0ea76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,6 +190,7 @@ if(ENABLE_OPENFAST)
   target_link_libraries(nalu PUBLIC ${OpenFAST_LIBRARIES} ${OpenFAST_CPP_LIBRARIES})
   target_include_directories(nalu SYSTEM PUBLIC ${OpenFAST_INCLUDE_DIRS})
   target_compile_definitions(nalu PUBLIC NALU_USES_OPENFAST)
+  enable_language(Fortran)
 endif()
 
 ############################ TIOGA #####################################
