diff --git a/include/master_element/Hex8FEM.h b/include/master_element/Hex8FEM.h
index 3eac756..0b9263c 100644
--- a/include/master_element/Hex8FEM.h
+++ b/include/master_element/Hex8FEM.h
@@ -84,7 +84,7 @@ private:
                                     +glIP, +glIP, +glIP, -glIP, +glIP, +glIP};
 
   // standard integration location +/ sqrt(3)/3
-  const double gIP = std::sqrt(3.0) / 3.0;
+  const double gIP = 0.577350269189626; //std::sqrt(3.0) / 3.0;
   const double intgLoc_[numIntPoints_ * nDim_] = {
     -gIP, -gIP, -gIP, +gIP, -gIP, -gIP, +gIP, +gIP, -gIP, -gIP, +gIP, -gIP,
     -gIP, -gIP, +gIP, +gIP, -gIP, +gIP, +gIP, +gIP, +gIP, -gIP, +gIP, +gIP};
